ngraph_bridge/ngraph_builder.cc

```diff
@@ -1977,13 +1977,27 @@ static Status TranslateFillOp(
 static Status TranslateFloorDivOp(
     const Node* op, const std::vector<const Tensor*>& static_input_map,
     Builder::OpMap& ng_op_map) {
-  auto ng_floordiv = [&op](std::shared_ptr<ng::Node> ng_input1,
-                           std::shared_ptr<ng::Node> ng_input2) {
-    return ConstructNgNode<ng::op::Floor>(
-        op->name(),
-        ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2));
-  };
-  return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_floordiv);
+  DataType dtype;
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "T", &dtype));
+  std::initializer_list<DataType> int_types = {DT_INT8, DT_UINT16, DT_INT16,
+                                               DT_INT32, DT_INT64};
+  if (std::find(int_types.begin(), int_types.end(), dtype) != int_types.end()) {
+    auto ng_div = [&op](std::shared_ptr<ng::Node> ng_input1,
+                        std::shared_ptr<ng::Node> ng_input2) {
+      return ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2);
+    };
+    return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_div);
+  } else {
+    auto ng_floordiv = [&op](std::shared_ptr<ng::Node> ng_input1,
+                             std::shared_ptr<ng::Node> ng_input2) {
+      return ConstructNgNode<ng::op::Floor>(
+          op->name(),
+          ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2));
+    };
+    return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_floordiv);
+  }
+  return errors::Internal(
```

Comment:
Please remove this error
" We should never reach this part in translating floordiv");"

And move this out of the if-else
return TranslateBinaryOp(op, static_input_map, ng_op_map, ng_floordiv);